Add override mark threshold feature to VxScan election manager screen

taravancil opened the pull request:
## Overview
Closes #1856 

This adds the same override mark thresholds modal from VxCentralScan to the election manager screen in VxScan.

## Demo Video or Screenshot

https://user-images.githubusercontent.com/7584833/184223762-266bb150-f333-4efd-80f3-e75804890763.mov

## Testing Plan 
Manual testing plus automated tests.

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports
<!-- for user-facing changes, non-user facing changes can remove or ignore the below items -->
- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

beausmith commented:
@taravancil - What keyboard is used for entering numbers?

taravancil commented:
@beausmith oh shoot I didn't even think about that 😭

arsalansufi commented on frontends/precinct-scanner/src/components/set_mark_thresholds_modal.tsx:
```diff
@@ -0,0 +1,237 @@
+  definite: 0.25,
+};
+
+export function SetMarkThresholdsModal({
```
Definitely a post-m14 thought that doesn't need to be addressed now! Just noting that we could move this component to a shared location since it's identical across VxScan and VxCentralScan (I know that's going to change a bit once we account for the lack of a keyboard on VxScan)

arsalansufi reviewed (commented):
All looking great! Will review again once the lack-of-keyboard issue has been addressed

taravancil commented:
@arsalansufi this is ready for review now! I updated the `input` type to be `number` and added a `step` of `.005`. Lmk if you think that should be changed.

arsalansufi commented on frontends/precinct-scanner/src/components/set_mark_thresholds_modal.test.tsx:
```diff
@@ -0,0 +1,171 @@
+  );
+  getByText('Override Mark Thresholds');
+  getByText(/WARNING: Do not proceed/);
+  fireEvent.click(getByText('Close'));
```
I know this is copied over from frontends/bsd so not a big deal! We could use `screen` queries instead of the ones returned by `render` and `userEvent` instead of `fireEvent` to follow the latest practices:

```ts
screen.getByText('Override Mark Thresholds');
screen.getByText(/WARNING: Do not proceed/);
userEvent.click(screen.getByText('Close'));
```

arsalansufi reviewed (approved):
Would love to see a screenshot of the new numerical inputs! Otherwise LGTM!

beausmith commented:
+1 - Please post screenshots or better videos of features! Especially when the feature changes and the video in the PR is not what is implemented. 😝
